k2/csrc/intersect.cu

```diff
@@ -90,950 +69,840 @@ using namespace intersect_internal;  // NOLINT
 // Caution: this is really a .cu file.  It contains mixed host and device code.
 
 /*
-   Pruned intersection (a.k.a. composition) that corresponds to decoding for
-   speech recognition-type tasks.  Can use either different decoding graphs (one
-   per acoustic sequence) or a shared graph
+   Intersection (a.k.a. composition) that corresponds to decoding for
+   speech recognition-type tasks.  This version does only forward-backward
+   pruning in the backward pass; the forward pass does no pruning.
+
+   Can use either different decoding graphs (one per acoustic sequence) or a
+   shared graph.
+
+   How to use this object:
+       Construct it
+       Call Intersect()
+       Call FormatOutput()
 */
 class MultiGraphDenseIntersect {
  public:
   /**
-     Pruned intersection (a.k.a. composition) that corresponds to decoding for
+     Intersection (a.k.a. composition) that corresponds to decoding for
      speech recognition-type tasks
 
        @param [in] a_fsas  The decoding graphs, one per sequence.  E.g. might
                            just be a linear sequence of phones, or might be
                            something more complicated.  Must have either the
-                           same Dim0() as b_fsas, or Size0()==1 in which
+                           same Dim0() as b_fsas, or Dim0() == 1 in which
```

Comment:
`Dim0() == 1` can be removed. The implementation assumes that a_fsas.Dim0() == b_fsas_.shape.Dim0().